x/ecocredit/server/basket/msg_create.go

```diff
@@ -19,31 +19,56 @@ func (k Keeper) Create(ctx context.Context, msg *basket.MsgCreate) (*basket.MsgC
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 
 	var fee sdk.Coins
-	k.paramsKeeper.Get(sdkCtx, core.KeyBasketCreationFee, &fee)
-	if !msg.Fee.IsAllGTE(fee) {
-		return nil, sdkerrors.ErrInsufficientFee.Wrapf("minimum fee %s, got %s", fee, msg.Fee)
-	}
+	k.paramsKeeper.Get(sdkCtx, core.KeyBasketFee, &fee)
 
 	curator, err := sdk.AccAddressFromBech32(msg.Curator)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, coin := range fee {
-		curatorBalance := k.bankKeeper.GetBalance(sdkCtx, curator, coin.Denom)
-		if curatorBalance.IsNil() || curatorBalance.IsLT(coin) {
-			return nil, sdkerrors.ErrInsufficientFunds.Wrapf("insufficient balance for bank denom %s", coin.Denom)
+	// In the next version of the basket package, this field will be updated to
+	// a single Coin rather than a list of Coins. In the meantime, the message
+	// will fail basic validation if more than one Coin is provided and only the
+	// minimum fee is checked against the balance of the curator account, sent
+	// to the basket submodule, and then burned by the basket submodule.
+	if len(fee) > 0 {
+
+		// check if single coin in msg.Fee is greater than or equal to any coin in fee
+		if !msg.Fee.IsAnyGTE(fee) {
+			if len(fee) > 1 {
+				return nil, sdkerrors.ErrInsufficientFee.Wrapf("minimum fee one of %s, got %s", fee, msg.Fee)
+			} else {
+				return nil, sdkerrors.ErrInsufficientFee.Wrapf("minimum fee %s, got %s", fee, msg.Fee)
+			}
 		}
-	}
 
-	err = k.bankKeeper.SendCoinsFromAccountToModule(sdkCtx, curator, basket.BasketSubModuleName, fee)
-	if err != nil {
-		return nil, err
-	}
+		var minimumFee sdk.Coin
 
-	err = k.bankKeeper.BurnCoins(sdkCtx, basket.BasketSubModuleName, fee)
-	if err != nil {
-		return nil, err
+		// find the minimum fee
+		for _, coin := range fee {
+			if coin.Denom == msg.Fee[0].Denom {
+				if minimumFee.IsNil() || coin.IsLT(minimumFee) {
+					minimumFee = coin
+				}
+			}
+		}
+
+		curatorBalance := k.bankKeeper.GetBalance(sdkCtx, curator, minimumFee.Denom)
+		if curatorBalance.IsNil() || curatorBalance.IsLT(minimumFee) {
+			return nil, sdkerrors.ErrInsufficientFunds.Wrapf("insufficient balance for bank denom %s", minimumFee.Denom)
+		}
+
+		minimumFees := sdk.NewCoins(minimumFee)
```

Comment:
might be better to just use `sdk.Coins{minimumFee}`  here, `NewCoins` runs all the validation again

Reply:
https://github.com/regen-network/regen-ledger/pull/1184/commits/44db6295c6bfd2bcb3d86e49260c7b599e312549